virttest/utils_libvirt/libvirt_ceph_utils.py

```diff
@@ -0,0 +1,154 @@
+    hotplug = "yes" == params.get("virt_disk_device_hotplug", "no")
+    keep_raw_image_as = "yes" == params.get("keep_raw_image_as", "no")
+
+    # Ceph related configurations
+    ceph_mon_ip = params.get("ceph_mon_ip", "EXAMPLE_MON_HOST")
+    ceph_host_port = params.get("ceph_host_port", "EXAMPLE_PORTS")
+    ceph_disk_name = params.get("ceph_disk_name", "EXAMPLE_SOURCE_NAME")
+    ceph_client_name = params.get("ceph_client_name")
+    ceph_client_key = params.get("ceph_client_key")
+    ceph_auth_user = params.get("ceph_auth_user")
+    ceph_auth_key = params.get("ceph_auth_key")
+    auth_sec_usage_type = params.get("ceph_auth_sec_usage_type", "ceph")
+    storage_size = params.get("storage_size", "1G")
+    img_file = params.get("ceph_image_file")
+    key_file = os.path.join(data_dir.get_tmp_dir(), "ceph.key")
+    key_opt = ""
+    is_local_img_file = True if img_file is None else False
+
+    # Prepare a blank params to confirm if delete the configure at the end of the test
+    ceph_cfg = ""
+    disk_auth_dict = None
+    if not utils_package.package_install(["ceph-common"]):
+        raise exceptions.TestError("Failed to install ceph-common")
+
+    # Create config file if it doesn't exist
+    ceph_cfg = ceph.create_config_file(ceph_mon_ip)
+    # If enable auth, prepare a local file to save key
+    if ceph_client_name and ceph_client_key:
+        with open(key_file, 'w') as f:
+            f.write("[%s]\n\tkey = %s\n" %
+                    (ceph_client_name, ceph_client_key))
+        key_opt = "--keyring %s" % key_file
+    if is_setup:
+        # If enable auth, prepare device source
+        if ceph_client_name and ceph_client_key:
+            auth_sec_dict = {"sec_usage": auth_sec_usage_type,
+                             "sec_name": "ceph_auth_secret"}
+            auth_sec_uuid = libvirt.create_secret(auth_sec_dict)
+            virsh.secret_set_value(auth_sec_uuid, ceph_auth_key,
+                                   debug=True)
+            disk_auth_dict = {"auth_user": ceph_auth_user,
+                              "secret_type": auth_sec_usage_type,
+                              "secret_uuid": auth_sec_uuid}
+            device_source = "rbd:%s:mon_host=%s:keyring=%s" % (ceph_disk_name,
+                                                               ceph_mon_ip,
+                                                               key_file)
+        else:
+            device_source = "rbd:%s:mon_host=%s" % (ceph_disk_name, ceph_mon_ip)
+        logging.debug("device source is: %s", device_source)
+        # clean up image file if exists
+        cmd = ("rbd -m {0} {1} info {2} && rbd -m {0} {1} rm "
+               "{2}".format(ceph_mon_ip, key_opt, ceph_disk_name))
+        process.run(cmd, ignore_status=True, shell=True)
+
+        #Create necessary image file if not exists
+        if img_file is None:
+            img_file = os.path.join(data_dir.get_tmp_dir(),
```

Comment:
Considering the current tmpfs situation, I strongly suggest we run this test on out latest product, to verify.

Reply:
Whether the 'tmp' dir is suitable or not in  the future

Reply:
That is good question. 
In any cases, we need one common utility data_dir.get_tmp_dir(), which quite large numbers of   case python file depends on heavily.
If data_dir.get_tmp_dir() has issue in new environment, the answer is clear and simple that we need fix in that, rather than painfully fix in each case python file

Reply:
Please refer to https://github.com/avocado-framework/avocado-vt/pull/2990
Or if you'd like to fix it another way.

Reply:
Agree to use get_data_dir especially on new written case

Reply:
updated